### PR TITLE
Fix add new report

### PR DESCRIPTION
**Pull request by Fosol**
Fixed the issue when saving a new report with a new template.

## Report Admin

image

### CODE REVIEW

File: api/net/Areas/Editor/Controllers/ContentController.cs [hunk 1]
```diff
@@ -290,13 +290,13 @@ public async Task<IActionResult> UpdateTopicsAsync(long id, IEnumerable<ContentT
 
         var updatedTopics = _contentService.AddOrUpdateContentTopics(id, topics.ToList().ConvertAll(x => x.ToEntity(id)));
 
-        var updatedContent = _contentService.FindById(id);
+        var updatedContent = _contentService.FindById(id) ?? throw new NoContentException("Failed to find content");
```
Comment (Fosol) [inline api/net/Areas/Editor/Controllers/ContentController.cs:293]: Handle `nullable`.  Need to use linting to see these some times

File: app/editor/src/features/admin/reports/ReportForm.tsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ const ReportForm: React.FC = () => {
 
   const [report, setReport] = React.useState<IReportModel>({
     ...defaultReport,
-    ownerId: userInfo?.id ?? 0,
+    ownerId: userInfo?.id,
```
Comment (Fosol) [inline app/editor/src/features/admin/reports/ReportForm.tsx:53]: Default to `undefined`.  On a page refresh it was getting set to `0` which was an invalid foreign key.

File: app/editor/src/features/admin/reports/constants/defaultRazorTemplate.ts [hunk 3]
```diff
@@ -1,35 +1,209 @@
-export const defaultRazorTemplate = `@inherits RazorEngineCore.RazorEngineTemplateBase<TNO.Services.Reporting.Models.ReportEngineContentModel>
+export const defaultRazorTemplate = `@inherits RazorEngineCore.RazorEngineTemplateBase<TNO.TemplateEngine.Models.Reports.ReportEngineContentModel>
```
Comment (Fosol) [inline app/editor/src/features/admin/reports/constants/defaultRazorTemplate.ts:1]: Update default template to match custom template

File: libs/net/dal/Services/AVOverviewInstanceService.cs [hunk 4]
```diff
@@ -29,7 +29,7 @@ public class AVOverviewInstanceService : BaseService<AVOverviewInstance, int>, I
             .Include(i => i.Template)
             .Include(i => i.Sections).ThenInclude(s => s.Source)
             .Include(i => i.Sections).ThenInclude(s => s.Series)
-            .Include(i => i.Sections).ThenInclude(s => s.Items).ThenInclude(i => i.Content).ThenInclude(c => c.FileReferences)
+            .Include(i => i.Sections).ThenInclude(s => s.Items).ThenInclude(i => i.Content).ThenInclude(c => c!.FileReferences)
```
Comment (Fosol) [inline libs/net/dal/Services/AVOverviewInstanceService.cs:32]: Handle `nullable`.  Need to use linting to see these some times

File: libs/net/dal/Services/ReportService.cs [hunk 5]
```diff
@@ -154,8 +154,8 @@ public override Report Add(Report entity)
             if (entity.Template.Id == 0)
             {
                 this.Context.Add(entity.Template);
-                this.Context.Add(entity.Template.ChartTemplates);
-                this.Context.Add(entity.Template.ChartTemplatesManyToMany);
+                this.Context.ChartTemplates.AddRange(entity.Template.ChartTemplates);
```
Comment (Fosol) [inline libs/net/dal/Services/ReportService.cs:157]: Resolve error when saving a new report with a new template.

**Review by kylgarmor — APPROVED** [review]

**Pull request merged**